R/utils.R

```diff
@@ -294,7 +295,7 @@ create_app_id <- function(data, modules) {
   checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
   checkmate::assert_class(modules, "teal_modules")
 
-  hashables <- list(data = data, modules = modules)
+  hashables <- c(data, modules)
```

Comment:
This will not work as the next line will fail to find the `$data` property

Reply:
Do. Not. Do this.

Reply:
https://github.com/insightsengineering/teal/issues/1058

Reply:
Ok, then the next expression needs to be changed somehow, (as the first condition in the if clause will never be true)

Nor will the `data` element be overwritten (it doesn't exist)

Reply:
Maybe this:

```r
  hashables_data <- if (inherits(data, "teal_data")) {
    as.list(data@env)
  } else if (inherits(data, "teal_data_module")) {
    body(data$server)
  }
  rlang::hash(c(hashables_data, modules))
```

Reply:
> Ok, then the next expression needs to be changed somehow, (as the first condition in the if clause will never be true)

Why not? 
```
  checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
```

Reply:
I do agree a clean-up is called for, though, I suggest:
```
create_app_id <- function(data, modules) {
  checkmate::assert_multi_class(data, c("teal_data", "teal_data_module"))
  checkmate::assert_class(modules, "teal_modules")

  data <- if (inherits(data, "teal_data")) {
    as.list(data@env)
  } else if (inherits(data, "teal_data_module")) {
    body(data$server)
  }
  rlang::hash(list(data = data, modules = modules))
}
```
I want to be absolutely sure it will be a list.

Reply:
ahhhh, I misunderstood the "Do. Not. Do. This." to be about my suggestion, not the PR code.

Yes it should be a list. I like your suggestion

Reply:
Please forgive my lack of clarity, this particular snippet has frustrated me greatly.

Reply:
> I do agree a clean-up is called for, though, I suggest:

thanks @averissimo  and @chlebowa 
This code snippet is not part of this PR,  and you have already fixed it in commit https://github.com/insightsengineering/teal/commit/d03cacc9977fad3a1df5e01b826e388293a63c4a. This may be the result of a bad merge from pre-release-cleanup@main. I will push your suggestion in pre-release-cleanup@main.

Reply:
Here [1ff3a13](https://github.com/insightsengineering/teal/pull/1021/commits/1ff3a1320a46e0a24bd45d33e514d2f9c55c5e63)